radixdlt-core/radixdlt/src/test/java/org/radix/api/jsonrpc/HighLevelApiHandlerTest.java

```diff
@@ -107,6 +116,43 @@ public void testTokenInfo() {
 		assertEquals(UInt256.EIGHT, result.get("currentSupply"));
 	}
 
+	@Test
+	public void testTransactionHistory() {
+		var service = mock(HighLevelApiService.class);
+		var handler = new HighLevelApiHandler(service);
+
+		var now = Instant.now();
+		var action = ActionEntry.unknown();
+		var entry = TxHistoryEntry.create(
+			AID.ZERO, now, UInt256.ONE, MessageEntry.create("text", "scheme"), List.of(action)
+		);
+
+		when(service.getTransactionHistory(any(), eq(5), any()))
+			.thenReturn(Result.ok(tuple(Optional.ofNullable(now), List.of(entry))));
+
+		var params = jsonObject().put("address", KNOWN_ADDRESS_STRING).put("size", 5);
+		var response = handler.handleTransactionHistory(jsonObject().put("id", "1").put("params", params));
+
+		assertNotNull(response);
+
+		var result = response.getJSONObject("result");
+
+		assertTrue(result.has("cursor"));
+		assertTrue(result.has("transactions"));
+		var transactions = result.getJSONArray("transactions");
+		assertEquals(1, transactions.length());
+
+		var singleTransaction = transactions.getJSONObject(0);
```

Comment:
Assert `"sentAt"` date field and `"txID"` field? :)

Reply:
updated